### ISSUE TITLE
Question on the report

### ISSUE THREAD

**Issue by jarbel** [NilsBaillie/BML-MCVAE#1]
Thanks for the report and repo. I have a question about your experiments: could you be more specific on the decoder distributions you chose in cases 1(Laplace) and 2 (student-t)?
Best
Julyan

**Comment by NilsBaillie** [comment]
Hello, for the distributions in the decoder we did as follows,
when the Gaussian distribution was used for instance in the definition of the class Toy : 
torch.distributions.Normal(loc=2 * np.pi * (torch.sqrt(torch.sum(torch.pow(z, 2), dim=1, keepdim=True)) + 2) + 0. * self.decoder_net.aux, scale=self.decoder_net.sigma).log_prob(x).sum(-1)

We replaced this term by :
torch.distributions.Laplace(loc=2 * np.pi * (torch.sqrt(torch.sum(torch.pow(z, 2), dim=1, keepdim=True)) + 2) + 0. * self.decoder_net.aux,  scale=self.decoder_net.sigma).log_prob(x).sum(-1)

and :
torch.distributions.StudentT(df = df,loc=2 * np.pi * (torch.sqrt(torch.sum(torch.pow(z, 2), dim=1, keepdim=True)) + 2) + 0. * self.decoder_net.aux,  scale=self.decoder_net.sigma).log_prob(x).sum(-1)

We did not change the 'loc' and 'scale' parameters so that the position of the center and the scaling of the distributions
remain the same for all cases, since what we wanted to do, is to answer the questions :
- What happens with a similar distribution to the Gaussian but with a spike at the center (ie Laplace) ?
- What happens with a similar distribution to the Gaussian but with heavier tails (ie Student with df = 2) ? 

Since changing the distribution of the decoder changes the joint distribution, we modified the code accordingly. 
The encoder is the same neural network in all cases however.
